### PR TITLE
[system-command-line] remove unused code

### PR DESCRIPTION
**Pull request by vlada-shubina**
PR removes unused code after migration to new parser

### CODE REVIEW

File: test/dotnet-new3.UnitTests/DotnetNewSearch.cs [hunk 1]
```diff
@@ -871,7 +871,7 @@ private List<List<string>> ParseTableOutput(string stdOut, string[] expectedColu
             int headerLineIndex = Array.FindIndex(lines, line => expectedColumns.All(column => line.Contains(column)));
             string headerLine = lines[headerLineIndex];
             //table ends after empty line
-            int lastLineIndex = Array.FindIndex(lines, headerLineIndex + 1, line => line.Length == 0) - 1;
+            int lastLineIndex = Array.FindIndex(lines, headerLineIndex + 1, line => (line.Length == 0 || line.Contains("[Debug]"))) - 1;
```
Comment (bekir-ozturk) [inline test/dotnet-new3.UnitTests/DotnetNewSearch.cs:874]: What is this change for?

**Review by bekir-ozturk — APPROVED** [review]

File: test/dotnet-new3.UnitTests/DotnetNewSearch.cs (hunk 1 above)
Comment (vlada-shubina) [reply test/dotnet-new3.UnitTests/DotnetNewSearch.cs:874]: this is for https://github.com/dotnet/templating/pull/4134

this commit will vanish after rebase to main.

**Comment by vlada-shubina** [comment]
Added 2 more test fixes:
- https://github.com/dotnet/templating/pull/4132/commits/04c82109adfa78cdb996a994a164997fea9a008d - reason for failure in debug:rebuild-cache test, again `TemplatePackageManager` task was not awaited before instance disposal
- https://github.com/dotnet/templating/pull/4132/commits/e8794c59083cfc6de6c0ef85713a71a4ddcc4765 from https://github.com/dotnet/templating/pull/4133 - the output from dotnet search containing Unicode chars was not properly parsed otherwise (again proves importance of this fix).

File: src/dotnet-new3/Program.cs [hunk 2]
```diff
@@ -23,6 +23,11 @@ public static class Program
 
         public static int Main(string[] args)
         {
+            if (Console.IsOutputRedirected)
+            {
+                Console.OutputEncoding = System.Text.Encoding.UTF8;
```
Comment (DavidKarlas) [inline src/dotnet-new3/Program.cs:28]: I guess it's a bit more than just removal :P

File: src/Microsoft.TemplateEngine.Cli/Alias/AliasSupport.cs [hunk 3]
```diff
@@ -1,204 +1,203 @@
...
-                {
-                    aliasTokens.Add(token);
-                }
-            }
-
-            return aliasTokens;
-        }
-    }
-}
+//#nullable enable
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.Cli/Alias/AliasSupport.cs:4]: I would prefer if we just removed whole file, and look at history when we want to re-introduce functionality.
Just my 2 cents...

**Review by DavidKarlas — APPROVED** [review]

File: src/dotnet-new3/Program.cs (hunk 2 above)
Comment (vlada-shubina) [reply src/dotnet-new3/Program.cs]: have to added it because the tests were failing without it

**Pull request merged**